controlplane/internal/controllers/rke2controlplane_controller.go

```diff
@@ -623,3 +622,25 @@ func (r *RKE2ControlPlaneReconciler) ClusterToRKE2ControlPlane(o client.Object)
 
 	return nil
 }
+
+func getIPAddress(machine clusterv1.Machine) (ip string, err error) {
+
+	for _, address := range machine.Status.Addresses {
+		switch address.Type {
+		case "InternalIP":
```

Comment:
We can use the constants defined in CAPI for these: https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/common_types.go#L145:L146

Reply:
Good catch, thanks a lot! 
I updated it.